packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs

```diff
@@ -109,6 +111,21 @@ public void SetupConnection(bool console = false)
                     };
                 }
             });
+
+            _connection.On<string, bool>("checkInternetAccess", request =>
+            {
+                var httpService = _services.GetRequiredService<IHttpService>();
+                try
+                {
+                    var fileToDownload = "newtonsoft.json.json.gz";
+                    using Stream stream = httpService.DownloadS3FileAsync(fileToDownload).Result;
+                    return true;
```

Comment:
Do we need a retry mechanism here with a default number of retries?

Reply:
The httpService integrate with retry in the DI